persia/ctx.py

```diff
@@ -240,21 +231,22 @@ class TrainCtx(BaseCtx):
 
     def __init__(
         self,
-        grad_scaler: torch.cuda.amp.GradScaler = None,
+        grad_scaler: Optional[torch.cuda.amp.GradScaler] = None,
```

Comment:
Better to move all optional args behind to the required?

Reply:
If the `CtxStatus` is `CtxStatus.EVAL` or `CtxStatus.INFERENCE`, there may not required the grad_scalar